model/src/main/java/org/projectnessie/api/http/HttpContentsApi.java

```diff
@@ -47,7 +48,87 @@
 @Path("contents")
 public interface HttpContentsApi extends ContentsApi {
 
-  /** Get the properties of an object. */
+  /**
+   * This operation returns the {@link Contents} for a {@link ContentsKey} in a named-reference (a
+   * {@link org.projectnessie.model.Branch} or {@link org.projectnessie.model.Tag}).
+   *
+   * <p>If the table-metadata is tracked globally (Iceberg), Nessie returns a {@link Contents}
+   * object, that contains the most up-to-date part for the globally tracked part (Iceberg:
+   * table-metadata) plus the per-Nessie-reference/hash specific part (Iceberg: snapshot-ID).
+   *
+   * @param key the {@link ContentsKey}s to retrieve
+   * @param ref named-reference to retrieve the contents for
+   * @param hashOnRef hash on {@code ref} to retrieve the contents for, translates to {@code HEAD},
+   *     if missing/{@code null}
+   * @return list of {@link org.projectnessie.model.MultiGetContentsResponse.ContentsWithKey}s
+   * @throws NessieNotFoundException if {@code ref} or {@code hashOnRef} does not exist
+   */
+  @GET
+  @Produces(ApiVersions.APPLICATION_NESSIE_GLOBAL_STATE)
+  @Path("{key}")
+  @Operation(
+      summary = "Get object content associated with a key.",
+      description =
+          "This operation returns the contents-value for a contents-key in a named-reference "
+              + "(a branch or tag).\n"
+              + "\n"
+              + "If the table-metadata is tracked globally (Iceberg), "
+              + "Nessie returns a 'Contents' object, that contains the most up-to-date part for "
+              + "the globally tracked part (Iceberg: table-metadata) plus the "
+              + "per-Nessie-reference/hash specific part (Iceberg: snapshot-ID).")
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Information for table",
+        content =
+            @Content(
+                mediaType = ApiVersions.APPLICATION_NESSIE_GLOBAL_STATE,
+                examples = {@ExampleObject(ref = "iceberg")},
+                schema = @Schema(implementation = Contents.class))),
+    @APIResponse(responseCode = "400", description = "Invalid input, ref name not valid"),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(
+        responseCode = "403",
+        description = "Not allowed to view the given reference or read object content for a key"),
+    @APIResponse(responseCode = "404", description = "Table not found on ref")
+  })
+  Contents getContentsWithGlobalState(
```

Comment:
why have two versions of get Contents? They seem the same to me? Why does the client care if its global or not

Reply:
Backwards compatibilty

Reply:
with what? I don't think we should try to maintain backwards compatibility now that our last pre-1.0 release is out

Reply:
Okay, that's new information.

Reply:
do you disagree? Looking at these changes it seems like it would be hard to support both pre-1.0 and 1.0 apis well.

It means Iceberg wont be able to use Nessie > 1,0 until the next release of Iceberg, which is a bummer but do you think it can be avoided?

Reply:
Lets discuss this today...we should try and keep backwards compatibility if its not too onerous

Reply:
It's confusing: _"I don't think we should try to maintain backwards compatibility"_ vs _"we should try and keep backwards compatibility"_

Reply:
yes it is confusing. How do we make it less confusing? What do you think in terms of backwards compatibility? 

For me there are competing concerns:
* Part of me doesn't think we should expend the effort as we are explicitly going to 1.0 and have no compatibiliy guarantees. 
* Im also concerned that Iceberg and other tools can't use Nessie 1.0 until their next releases (possibly months?)
* Iceberg in particular probably shoudln't use pre-1.0 Nessie since we can't do migrations of that data structure

Reply:
I think, a "clear cut" for 1.0 is the better way forwards, because probably nobody is using Nessie in production yet. The argument that current Iceberg (0.12.0) wouldn't be able to run against Nessie 1.0 is legit though - couple options there:
- either we can ask to release Iceberg 0.12.1 or (0.13.0?) w/ Nessie 1.0
- release a custom artifact `org.projectnessie:iceberg-nessie:0.12.1` for the time until Iceberg 0.13.0

Ensuring backwards compatibility forces us to add a bunch of complexity to the code base (think: `commitMultipleOperationsNessie09`, `getContentsNessie09`, a Nessie 0.9 version of `IcebergTable`, a Nessie 1.0 version of `IcebergTable`). It may also mean that people continue to use Iceberg 0.12 w/ Nessie 1.0 but actually risking data corruption, because that doesn't know about global-states.